app/gui2/src/components/GraphEditor/widgets/WidgetTableEditor/TableHeader.vue

```diff
@@ -0,0 +1,88 @@
+<script lang="ts">
+import type { IHeaderParams } from 'ag-grid-community'
+import { ref, watch } from 'vue'
+
+/** Parameters recognized by this header component.
+ *
+ * They are set through `headerComponentParams` option in AGGrid column definition.
+ */
+export interface HeaderParams {
+  /** Setter called when column name is changed by the user. */
+  nameSetter?: (newName: string) => void
+  /** Column is virtual if it is not represented in the AST. Such column might be used
+   * to create new one.
+   */
+  virtualColumn?: boolean
+  onHeaderEditingStarted?: (stop: (cancel: boolean) => void) => void
+  onHeaderEditingStopped?: () => void
+}
+</script>
+
+<script setup lang="ts">
+const props = defineProps<{
+  params: IHeaderParams & HeaderParams
+}>()
+
+const editing = ref(false)
+const inputElement = ref<HTMLInputElement>()
+
+watch(editing, (newVal, oldVal) => {
+  if (newVal && !oldVal) {
```

Comment:
A `ref` watcher will not be called if the value hasn't changed.

```suggestion
  if (newVal) {
```